plugin/src/main/groovy/com/novoda/staticanalysis/internal/ktlint/KtlintConfigurator.groovy

```diff
@@ -84,12 +84,19 @@ class KtlintConfigurator implements Configurator {
     }
 
     private void configureAndroidWithVariants(def mainVariants) {
-        mainVariants.all { configureKtlint(it.name) }
-        variantFilter.filteredTestVariants.all { configureKtlint(it.name) }
-        variantFilter.filteredUnitTestVariants.all { configureKtlint(it.name) }
+        mainVariants.all { configureAndroidVariant(it) }
+        variantFilter.filteredTestVariants.all { configureAndroidVariant(it) }
+        variantFilter.filteredUnitTestVariants.all { configureAndroidVariant(it) }
     }
 
-    private void configureKtlint(def sourceSetName) {
+    private void configureAndroidVariant(def variant) {
+        variant.sourceSets.each { sourceSet ->
+            configureKtlint(sourceSet.name)
+        }
+        configureKtlint(variant.name)
```

Comment:
isn't this already covered by L93-L95? also, is it possible to use `.all {}` here instead?

Reply:
Don't know exactly why but KtLint plugin changed from supporting variants to sourceSets. That's why it was broken. So this two together kinda support all versions automatically. 

On a second thought. Maybe it is not necessary. I will try without it.

Reply:
I tried that but didn't work. 

Since we use variants.all, at this point, variant is resolved with all details. At this point sourceSets is not dynamic and does not have .all method.

Reply:
Fixed in https://github.com/novoda/gradle-static-analysis-plugin/pull/153/commits/be0bf8269351081ffcd8c184d258bc61508d6d34